lib/web/controllers/users_test.go

```diff
@@ -0,0 +1,288 @@
+package webcontrollers_test
+
+import (
+	"net/http"
+	"net/url"
+	"strconv"
+
+	"go-google-scraper-challenge/constants"
+	. "go-google-scraper-challenge/test"
+
+	"github.com/bxcodec/faker/v3"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+)
+
+var _ = Describe("UsersController", func() {
+	Describe("GET /signup", func() {
+		Context("given user is not signed in", func() {
+			It("renders with status 200", func() {
+				response := MakeWebRequest("GET", "/signup", nil, nil)
+
+				Expect(response.StatusCode).To(Equal(http.StatusOK))
+			})
+		})
+
+		// TODO: Test this when work on the result list screen
+		XContext("given user already signed in", func() {
+			It("redirects to root path", func() {
+				user := FabricateUser(faker.Email(), faker.Password())
+				response := MakeWebRequest("GET", "/signup", nil, user)
+
+				currentPath := GetCurrentPath(response)
+
+				Expect(response.StatusCode).To(Equal(http.StatusFound))
+				Expect(currentPath).To(Equal("/"))
+			})
+		})
+	})
+
+	Describe("POST /users", func() {
+		Context("given user not signed in", func() {
```

Comment:
Should it be
```suggestion
		Context("given a user that does not signed in", func() {
```
or
```suggestion
		Context("given an unauthenticated user", func() {
```

Reply:
Updated in [108b56e](https://github.com/carryall/go-google-scraper-challenge/pull/125/commits/108b56ea09e98755e04396d630575bfc4df11f86) 🙇🏼